pkg/collector/corechecks/ebpf/tcp_queue_length.go

```diff
@@ -0,0 +1,137 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-2019 Datadog, Inc.
+
+// +build linux
+
+package ebpf
+
+import (
+	"sort"
+	"strconv"
+	"strings"
+
+	yaml "gopkg.in/yaml.v2"
+
+	"github.com/DataDog/datadog-agent/pkg/aggregator"
+	"github.com/DataDog/datadog-agent/pkg/autodiscovery/integration"
+	"github.com/DataDog/datadog-agent/pkg/collector/check"
+	core "github.com/DataDog/datadog-agent/pkg/collector/corechecks"
+	process_net "github.com/DataDog/datadog-agent/pkg/process/net"
+	"github.com/DataDog/datadog-agent/pkg/tagger"
+	"github.com/DataDog/datadog-agent/pkg/tagger/collectors"
+	"github.com/DataDog/datadog-agent/pkg/util/containers"
+	"github.com/DataDog/datadog-agent/pkg/util/log"
+)
+
+const (
+	tcpQueueLengthCheckName = "tcp_queue_length"
+)
+
+// TCPQueueLengthConfig is the config of the TCP Queue Length check
+type TCPQueueLengthConfig struct {
+	CollectTCPQueueLength bool `yaml:"collect_tcp_queue_length"`
+	OnlyCountNbContexts   bool `yaml:"only_count_nb_contexts"` // For impact analysis only. To be removed after
+}
+
+// TCPQueueLengthCheck grabs TCP queue length metrics
+type TCPQueueLengthCheck struct {
+	core.CheckBase
+	instance *TCPQueueLengthConfig
+}
+
+func init() {
+	core.RegisterCheck(tcpQueueLengthCheckName, TCPQueueLengthFactory)
+}
+
+// TCPQueueLengthFactory is exported for integration testing
+func TCPQueueLengthFactory() check.Check {
+	return &TCPQueueLengthCheck{
+		CheckBase: core.NewCheckBase(tcpQueueLengthCheckName),
+		instance:  &TCPQueueLengthConfig{},
+	}
+}
+
+// Parse parses the check configuration and init the check
+func (t *TCPQueueLengthConfig) Parse(data []byte) error {
+	// default values
+	t.CollectTCPQueueLength = true
+	t.OnlyCountNbContexts = true
+
+	if err := yaml.Unmarshal(data, t); err != nil {
+		return err
+	}
+	return nil
+}
+
+//Configure parses the check configuration and init the check
+func (t *TCPQueueLengthCheck) Configure(config, initConfig integration.Data, source string) error {
+	// TODO: Remove that hard-code and put it somewhere else
+	process_net.SetSystemProbeSocketPath("/opt/datadog-agent/run/sysprobe.sock")
```

Comment:
FYI @stb2132  also needs to get at the remote socket path from the core agent for the purposes of generating a flare.

Reply:
In kubernetes, accessing the `system-probe` socket from the core agent requires some new mount volumes in containers. Let me find my old PRs about that.

Reply:
Here are the changes in the helm chart that are needed by this PR:
https://github.com/L3n41c/charts/commit/eeb47336fb755e6e445c196c2ce6c86b3d7de325
* share the UNIX domain socket between `system-probe` and the core agent
* mount the kernel headers of the host inside the `system-probe` container.

Reply:
Indeed.
I made the path configurable but with a default value.